lib/realtime/tenants/authorization/policies/broadcast_policies.ex

```diff
@@ -24,74 +23,64 @@ defmodule Realtime.Tenants.Authorization.Policies.BroadcastPolicies do
+      {:ok, []} ->
+        {:ok, Policies.update_policies(policies, :broadcast, :read, false)}
 
-        {:error, %Postgrex.Error{postgres: %{code: :insufficient_privilege}}} ->
-          Policies.update_policies(policies, :broadcast, :read, false)
+      {:ok, [%Message{}]} ->
+        {:ok, Policies.update_policies(policies, :broadcast, :read, true)}
 
-        {:error, :not_found} ->
-          Policies.update_policies(policies, :broadcast, :read, false)
+      {:error, %Postgrex.Error{postgres: %{code: :insufficient_privilege}}} ->
+        {:ok, Policies.update_policies(policies, :broadcast, :read, false)}
 
-        {:error, error} ->
-          log_error(
-            "UnableToSetPolicies",
-            "Error getting policies for connection: #{to_log(error)}"
-          )
+      {:error, error} ->
+        log_error(
+          "UnableToSetPolicies",
+          "Error getting policies for connection: #{to_log(error)}"
+        )
 
-          Postgrex.rollback(transaction_conn, error)
-      end
-    end)
+        Postgrex.rollback(conn, error)
+    end
   end
 
   @impl true
-  def check_write_policies(_conn, policies, %Authorization{channel: nil}) do
+  def check_write_policies(_conn, policies, %Authorization{channel_name: nil}) do
     {:ok, Policies.update_policies(policies, :broadcast, :write, false)}
   end
 
-  def check_write_policies(conn, policies, %Authorization{
-        channel: %Channel{id: channel_id}
-      }) do
-    Postgrex.transaction(conn, fn transaction_conn ->
-      query = from(b in Broadcast, where: b.channel_id == ^channel_id)
-
-      case Repo.one(conn, query, Broadcast, mode: :savepoint) do
-        {:ok, %Broadcast{} = broadcast} ->
-          zero = NaiveDateTime.new!(~D[1970-01-01], ~T[00:00:00])
-          changeset = Broadcast.check_changeset(broadcast, %{updated_at: zero})
-
-          case Repo.update(conn, changeset, Broadcast, mode: :savepoint) do
-            {:ok, %Broadcast{updated_at: ^zero}} ->
-              Postgrex.query!(transaction_conn, "ROLLBACK AND CHAIN", [])
-              Policies.update_policies(policies, :broadcast, :write, true)
-
-            {:error, %Postgrex.Error{postgres: %{code: :insufficient_privilege}}} ->
-              Policies.update_policies(policies, :broadcast, :write, false)
-
-            {:error, :not_found} ->
-              Policies.update_policies(policies, :broadcast, :write, false)
-
-            {:error, error} ->
-              log_error(
-                "UnableToSetPolicies",
-                "Error getting policies for connection: #{to_log(error)}"
-              )
-          end
-
-        {:error, :not_found} ->
-          Policies.update_policies(policies, :broadcast, :write, false)
-      end
-    end)
+  def check_write_policies(conn, policies, %Authorization{channel_name: channel_name}) do
+    changeset =
+      Message.changeset(%Message{}, %{
+        channel_name: channel_name,
+        feature: :broadcast,
+        event: "dummy"
```

Comment:
maybe add a default event name?

Reply:
yes, unsure of what to be honest 😓

Reply:
`check_write_policy` for now, later we might want this to also be something we check